Flaky duties tests

1. 

```
=== RUN   TestScheduler_Attester_Genesis_Reorg_Previous_Epoch_Transition_Indices_Changed
2024-08-07T14:13:21.143479Z	DEBUG	logger is ready	{"level": "debug", "encoder": "capital", "format": "console", "file_options": null}
2024-08-07T14:13:21.143668Z	INFO	TestScheduler_Attester_Genesis_Reorg_Previous_Epoch_Transition_Indices_Changed.DutyScheduler	duty scheduler started
2024-08-07T14:13:21.143701Z	DEBUG	TestScheduler_Attester_Genesis_Reorg_Previous_Epoch_Transition_Indices_Changed.DutyScheduler	no active validators for epoch	{"handler": "ATTESTER", "epoch": 1}
2024-08-07T14:13:21.143712Z	INFO	TestScheduler_Attester_Genesis_Reorg_Previous_Epoch_Transition_Indices_Changed.DutyScheduler	starting duty handler	{"handler": "ATTESTER"}
2024-08-07T14:13:21.143732Z	DEBUG	TestScheduler_Attester_Genesis_Reorg_Previous_Epoch_Transition_Indices_Changed.DutyScheduler	🛠 ticker event	{"handler": "ATTESTER", "epoch_slot_pos": "e1-s63-#32"}
2024-08-07T14:13:21.143752Z	DEBUG	TestScheduler_Attester_Genesis_Reorg_Previous_Epoch_Transition_Indices_Changed.DutyScheduler	🗂 got duties	{"handler": "ATTESTER", "count": 1, "epoch": 2, "duties": "ATTESTER-e2-s66-v1", "duration": "0.000013041"}
    scheduler_test.go:274: 
        	Error Trace:	/Users/bloxmac/GolandProjects/ssv/operator/duties/scheduler_test.go:274
        	            				/Users/bloxmac/GolandProjects/ssv/operator/duties/attester_genesis_test.go:393
        	Error:      	timed out waiting for duties to be fetched
        	Test:       	TestScheduler_Attester_Genesis_Reorg_Previous_Epoch_Transition_Indices_Changed
--- FAIL: TestScheduler_Attester_Genesis_Reorg_Previous_Epoch_Transition_Indices_Changed (0.20s)
```

2. 

```
=== RUN   TestScheduler_Committee_Indices_Changed_Attester_Only
2024-08-07T14:13:46.378985Z	DEBUG	logger is ready	{"level": "debug", "encoder": "capital", "format": "console", "file_options": null}
2024-08-07T14:13:46.379536Z	INFO	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	duty scheduler started
2024-08-07T14:13:46.379710Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🗂 got duties	{"handler": "ATTESTER", "count": 0, "epoch": 0, "duties": "", "duration": "0.00003475"}
2024-08-07T14:13:46.379779Z	INFO	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	starting duty handler	{"handler": "ATTESTER"}
2024-08-07T14:13:46.379970Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	👥 got duties	{"handler": "SYNC_COMMITTEE", "count": 0, "period": "p0", "duties": "", "duration": "0.000091041"}
2024-08-07T14:13:46.380022Z	INFO	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	starting duty handler	{"handler": "SYNC_COMMITTEE"}
2024-08-07T14:13:46.380112Z	INFO	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	starting duty handler	{"handler": "CLUSTER"}
2024-08-07T14:13:46.380191Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🛠 ticker event	{"handler": "ATTESTER", "epoch_slot_pos": "e0-s0-#1"}
2024-08-07T14:13:46.380323Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🛠 ticker event	{"handler": "SYNC_COMMITTEE", "period_epoch_slot_pos": "p0-e0-s0-#1"}
2024-08-07T14:13:46.380582Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🛠 ticker event	{"handler": "CLUSTER", "period_epoch_slot_pos": "p0-e0-s0-#1"}
2024-08-07T14:13:46.580795Z	INFO	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🔁 indices change received	{"handler": "SYNC_COMMITTEE", "period_epoch_slot_pos": "p0-e0-s0-#1"}
2024-08-07T14:13:46.580930Z	INFO	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🔁 indices change received	{"handler": "ATTESTER", "epoch_slot_pos": "e0-s0-#1"}
2024-08-07T14:13:46.782091Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🛠 ticker event	{"handler": "ATTESTER", "epoch_slot_pos": "e0-s1-#2"}
2024-08-07T14:13:46.782342Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🗂 got duties	{"handler": "ATTESTER", "count": 3, "epoch": 0, "duties": "ATTESTER-e0-s0-v1, ATTESTER-e0-s1-v2, ATTESTER-e0-s2-v3", "duration": "0.000112333"}
2024-08-07T14:13:46.782423Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🛠 ticker event	{"handler": "CLUSTER", "period_epoch_slot_pos": "p0-e0-s1-#2"}
2024-08-07T14:13:46.782676Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🛠 ticker event	{"handler": "SYNC_COMMITTEE", "period_epoch_slot_pos": "p0-e0-s1-#2"}
2024-08-07T14:13:46.783447Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	👥 got duties	{"handler": "SYNC_COMMITTEE", "count": 0, "period": "p0", "duties": "", "duration": "0.000732416"}
    scheduler.go:428: Unexpected call to *duties.MockDutyExecutor.ExecuteCommitteeDuty([0x14000222380 [207 151 173 238 219 89 224 91 253 115 162 180 194 168 136 87 8 196 244 247 12 132 198 75 39 18 14 114 171 115 59 114] 0x140004fd1e0]) at /Users/bloxmac/GolandProjects/ssv/operator/duties/scheduler.go:428 because: there are no expected calls of the method "ExecuteCommitteeDuty" for that receiver
2024-08-07T14:13:46.983872Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🛠 ticker event	{"handler": "ATTESTER", "epoch_slot_pos": "e0-s2-#3"}
2024-08-07T14:13:46.983919Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🛠 ticker event	{"handler": "CLUSTER", "period_epoch_slot_pos": "p0-e0-s2-#3"}
2024-08-07T14:13:46.983901Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🛠 ticker event	{"handler": "SYNC_COMMITTEE", "period_epoch_slot_pos": "p0-e0-s2-#3"}
2024-08-07T14:13:46.984106Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🏃 Executing duty	{"handler": "ATTESTER", "beacon_role": "AGGREGATOR", "committee_index": 0, "current_slot": "2", "slot": 2, "epoch": 0, "pubkey": "010205000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", "start_time_unix_milli": "1723040026984", "duty": {"Type":1,"PubKey":"0x010205000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","Slot":"2","ValidatorIndex":"3","CommitteeIndex":0,"CommitteeLength":0,"CommitteesAtSlot":0,"ValidatorCommitteeIndex":0,"ValidatorSyncCommitteeIndices":null}}
2024-08-07T14:13:47.035195Z	DEBUG	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	🏃 Executing committee duty	{"handler": "CLUSTER", "committee_id": "cf97adeedb59e05bfd73a2b4c2a8885708c4f4f70c84c64b27120e72ab733b72", "role": "COMMITTEE_RUNNER", "duties": "ATTESTER-e0-s2-v3", "current_slot": "2", "slot": 2, "epoch": 0, "start_time_unix_milli": "1723040026983", "duty": {"Slot":"2","ValidatorDuties":[{"Type":0,"PubKey":"0x010205000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","Slot":"2","ValidatorIndex":"3","CommitteeIndex":0,"CommitteeLength":0,"CommitteesAtSlot":0,"ValidatorCommitteeIndex":0,"ValidatorSyncCommitteeIndices":null}]}}
2024-08-07T14:13:47.035533Z	INFO	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	duty handler exited	{"handler": "CLUSTER"}
2024-08-07T14:13:47.035567Z	INFO	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	duty handler exited	{"handler": "SYNC_COMMITTEE"}
2024-08-07T14:13:47.035577Z	INFO	TestScheduler_Committee_Indices_Changed_Attester_Only.DutyScheduler	duty handler exited	{"handler": "ATTESTER"}
--- FAIL: TestScheduler_Committee_Indices_Changed_Attester_Only (0.66s)
```

3.

```
=== RUN   TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only
2024-08-07T14:13:48.347597Z	DEBUG	logger is ready	{"level": "debug", "encoder": "capital", "format": "console", "file_options": null}
2024-08-07T14:13:48.347774Z	INFO	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	duty scheduler started
2024-08-07T14:13:48.347832Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🗂 got duties	{"handler": "ATTESTER", "count": 0, "epoch": 1, "duties": "", "duration": "0.0000105"}
2024-08-07T14:13:48.347863Z	INFO	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	starting duty handler	{"handler": "ATTESTER"}
2024-08-07T14:13:48.347954Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	👥 got duties	{"handler": "SYNC_COMMITTEE", "count": 0, "period": "p0", "duties": "", "duration": "0.000034959"}
2024-08-07T14:13:48.347986Z	INFO	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	starting duty handler	{"handler": "SYNC_COMMITTEE"}
2024-08-07T14:13:48.348036Z	INFO	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	starting duty handler	{"handler": "CLUSTER"}
2024-08-07T14:13:48.348112Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🛠 ticker event	{"handler": "ATTESTER", "epoch_slot_pos": "e1-s63-#32"}
2024-08-07T14:13:48.348160Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🛠 ticker event	{"handler": "CLUSTER", "period_epoch_slot_pos": "p0-e1-s63-#32"}
2024-08-07T14:13:48.348158Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🛠 ticker event	{"handler": "SYNC_COMMITTEE", "period_epoch_slot_pos": "p0-e1-s63-#32"}
2024-08-07T14:13:48.348234Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🗂 got duties	{"handler": "ATTESTER", "count": 1, "epoch": 2, "duties": "ATTESTER-e2-s66-v1", "duration": "0.000080417"}
2024-08-07T14:13:48.348281Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only	🏁 Head event: Block arrived before 1/3 slot	{"epoch_slot_pos": "e1-s63-#32", "time_saved": "50.00925ms"}
2024-08-07T14:13:48.549625Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🛠 ticker event	{"handler": "ATTESTER", "epoch_slot_pos": "e2-s64-#1"}
2024-08-07T14:13:48.549621Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🛠 ticker event	{"handler": "CLUSTER", "period_epoch_slot_pos": "p0-e2-s64-#1"}
2024-08-07T14:13:48.549656Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🛠 ticker event	{"handler": "SYNC_COMMITTEE", "period_epoch_slot_pos": "p0-e2-s64-#1"}
2024-08-07T14:13:48.749991Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only	🔀 Previous duty dependent root has changed on epoch transition	{"epoch_slot_pos": "e2-s64-#1", "old_current_dependent_root": "0x0100000000000000000000000000000000000000000000000000000000000000", "new_previous_dependent_root": "0x0200000000000000000000000000000000000000000000000000000000000000"}
2024-08-07T14:13:48.750168Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only	🏁 Head event: Block arrived before 1/3 slot	{"epoch_slot_pos": "e2-s64-#1", "time_saved": "50.026625ms"}
2024-08-07T14:13:48.750362Z	INFO	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🔀 reorg event received	{"handler": "SYNC_COMMITTEE", "period_epoch_slot_pos": "p0-e2-s64-#1", "event": {"Slot":"64","Previous":true,"Current":false}}
2024-08-07T14:13:48.750456Z	INFO	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🔀 reorg event received	{"handler": "ATTESTER", "epoch_slot_pos": "e2-s64-#1", "event": {"Slot":"64","Previous":true,"Current":false}}
2024-08-07T14:13:48.751518Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🛠 ticker event	{"handler": "CLUSTER", "period_epoch_slot_pos": "p0-e2-s65-#2"}
2024-08-07T14:13:48.751578Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🗂 got duties	{"handler": "ATTESTER", "count": 1, "epoch": 2, "duties": "ATTESTER-e2-s65-v1", "duration": "0.001026583"}
2024-08-07T14:13:48.751716Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🛠 ticker event	{"handler": "SYNC_COMMITTEE", "period_epoch_slot_pos": "p0-e2-s65-#2"}
2024-08-07T14:13:48.751918Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🛠 ticker event	{"handler": "ATTESTER", "epoch_slot_pos": "e2-s65-#2"}
2024-08-07T14:13:48.752169Z	DEBUG	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only.DutyScheduler	🏃 Executing duty	{"handler": "ATTESTER", "beacon_role": "AGGREGATOR", "committee_index": 0, "current_slot": "65", "slot": 65, "epoch": 2, "pubkey": "010203000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", "start_time_unix_milli": "1723040028752", "duty": {"Type":1,"PubKey":"0x010203000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","Slot":"65","ValidatorIndex":"1","CommitteeIndex":0,"CommitteeLength":0,"CommitteesAtSlot":0,"ValidatorCommitteeIndex":0,"ValidatorSyncCommitteeIndices":null}}
    scheduler_test.go:396: 
        	Error Trace:	/Users/bloxmac/GolandProjects/ssv/operator/duties/scheduler_test.go:396
        	            				/Users/bloxmac/GolandProjects/ssv/operator/duties/committee_test.go:677
        	Error:      	timed out waiting for duty to be executed
        	Test:       	TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only
--- FAIL: TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only (0.61s)
```

--- a/operator/duties/attester_genesis_test.go
+++ b/operator/duties/attester_genesis_test.go
@@ -190,8 +190,8 @@ func TestScheduler_Attester_Genesis_Indices_Changed(t *testing.T) {
 
 	// STEP 3: wait for attester duties to be fetched again
 	currentSlot.Set(phase0.Slot(1))
-	mockTicker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	mockTicker.Send(currentSlot.Get())
 	waitForGenesisDutiesFetch(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 	// no execution should happen in slot 1
 	waitForNoActionGenesis(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
@@ -254,8 +254,8 @@ func TestScheduler_Attester_Genesis_Multiple_Indices_Changed_Same_Slot(t *testin
 
 	// STEP 5: wait for attester duties to be fetched
 	currentSlot.Set(phase0.Slot(2))
-	mockTicker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	mockTicker.Send(currentSlot.Get())
 	waitForGenesisDutiesFetch(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
 	// STEP 6: wait for attester duties to be executed
@@ -304,8 +304,8 @@ func TestScheduler_Attester_Genesis_Reorg_Previous_Epoch_Transition(t *testing.T
 	})
 
 	// STEP 1: wait for attester duties to be fetched for next epoch
-	mockTicker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	mockTicker.Send(currentSlot.Get())
 	waitForGenesisDutiesFetch(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
 	// STEP 2: trigger head event
@@ -388,8 +388,8 @@ func TestScheduler_Attester_Genesis_Reorg_Previous_Epoch_Transition_Indices_Chan
 	})
 
 	// STEP 1: wait for attester duties to be fetched for next epoch
-	mockTicker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	mockTicker.Send(currentSlot.Get())
 	waitForGenesisDutiesFetch(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 	waitForNoActionGenesis(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
@@ -497,8 +497,8 @@ func TestScheduler_Attester_Genesis_Reorg_Previous(t *testing.T) {
 	waitForNoActionGenesis(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
 	// STEP 3: Ticker with no action
-	currentSlot.Set(phase0.Slot(33))
 	waitForDuties.Set(true)
+	currentSlot.Set(phase0.Slot(33))
 	mockTicker.Send(currentSlot.Get())
 	waitForNoActionGenesis(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 

--- a/operator/duties/attester_test.go
+++ b/operator/duties/attester_test.go
@@ -184,8 +184,8 @@ func TestScheduler_Attester_Indices_Changed(t *testing.T) {
 
 	// STEP 3: wait for attester duties to be fetched again
 	currentSlot.Set(phase0.Slot(1))
-	mockTicker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	mockTicker.Send(currentSlot.Get())
 	waitForDutiesFetch(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 	// no execution should happen in slot 1
 	waitForNoAction(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
@@ -248,8 +248,8 @@ func TestScheduler_Attester_Multiple_Indices_Changed_Same_Slot(t *testing.T) {
 
 	// STEP 5: wait for attester duties to be fetched
 	currentSlot.Set(phase0.Slot(2))
-	mockTicker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	mockTicker.Send(currentSlot.Get())
 	waitForDutiesFetch(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
 	// STEP 6: wait for attester duties to be executed
@@ -298,8 +298,8 @@ func TestScheduler_Attester_Reorg_Previous_Epoch_Transition(t *testing.T) {
 	})
 
 	// STEP 1: wait for attester duties to be fetched for next epoch
-	mockTicker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	mockTicker.Send(currentSlot.Get())
 	waitForDutiesFetch(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
 	// STEP 2: trigger head event

--- a/operator/duties/committee_test.go
+++ b/operator/duties/committee_test.go
@@ -2,7 +2,6 @@ package duties
 
 import (
 	"context"
-	"runtime"
 	"testing"
 	"time"
 
@@ -608,8 +607,6 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only_3(t *testing.T) {
 
 // reorg previous dependent root changed
 func TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only(t *testing.T) {
-	originalProcs := runtime.GOMAXPROCS(1)
-	defer runtime.GOMAXPROCS(originalProcs)
 	var (
 		dutyStore     = dutystore.New()
 		attHandler    = NewAttesterHandler(dutyStore.Attester)

--- a/operator/duties/committee_test.go
+++ b/operator/duties/committee_test.go
@@ -400,8 +400,8 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only(t *testing.T) {
 	require.Less(t, scheduler.network.Beacon.SlotDurationSec()/3, time.Since(startTime))
 
 	// Stop scheduler & wait for graceful exit.
-	cancel()
 	require.NoError(t, schedulerPool.Wait())
+	cancel()
 }
 
 func TestScheduler_Committee_Indices_Changed_Attester_Only_2(t *testing.T) {

--- a/operator/duties/committee_test.go
+++ b/operator/duties/committee_test.go
@@ -377,8 +377,8 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only(t *testing.T) {
 
 	// STEP 3: wait for attester duties to be fetched
 	currentSlot.Set(phase0.Slot(1))
-	ticker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	ticker.Send(currentSlot.Get())
 	// wait for attester duties to be fetched
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 	// wait for sync committee duties to be fetched
@@ -387,11 +387,10 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only(t *testing.T) {
 	waitForNoActionCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
 	// STEP 4: wait for committee duties to be executed
+	currentSlot.Set(phase0.Slot(2))
 	aDuties, _ := attDuties.Get(0)
-	const executionSlot = phase0.Slot(2)
-	committeeMap := commHandler.buildCommitteeDuties([]*eth2apiv1.AttesterDuty{aDuties[2]}, nil, 0, executionSlot)
+	committeeMap := commHandler.buildCommitteeDuties([]*eth2apiv1.AttesterDuty{aDuties[2]}, nil, 0, currentSlot.Get())
 	setExecuteDutyFuncs(scheduler, executeDutiesCall, len(committeeMap))
-	currentSlot.Set(executionSlot)
 
 	startTime := time.Now()
 	ticker.Send(currentSlot.Get())
@@ -476,8 +475,8 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only_2(t *testing.T) {
 
 	// STEP 3: wait for attester duties to be fetched
 	currentSlot.Set(phase0.Slot(1))
-	ticker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	ticker.Send(currentSlot.Get())
 	// wait for attester duties to be fetched
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 	// wait for sync committee duties to be fetched
@@ -564,8 +563,8 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only_3(t *testing.T) {
 
 	// STEP 3: wait for attester duties to be fetched
 	currentSlot.Set(phase0.Slot(1))
-	ticker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	ticker.Send(currentSlot.Get())
 	// wait for attester duties to be fetched
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 	// wait for sync committee duties to be fetched
@@ -629,8 +628,8 @@ func TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only(t *te
 	})
 
 	// STEP 1: wait for attester duties to be fetched for next epoch
-	ticker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	ticker.Send(currentSlot.Get())
 	// wait for attester duties to be fetched
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
@@ -733,8 +732,8 @@ func TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Indices_Changed_Att
 	})
 
 	// STEP 1: wait for attester duties to be fetched for next epoch
-	ticker.Send(currentSlot.Get())
 	waitForDuties.Set(true)
+	ticker.Send(currentSlot.Get())
 	// wait for attester duties to be fetched
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 

--- a/operator/duties/committee_test.go
+++ b/operator/duties/committee_test.go
@@ -405,8 +405,8 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only(t *testing.T) {
 	require.Less(t, scheduler.network.Beacon.SlotDurationSec()/3, time.Since(startTime))
 
 	// Stop scheduler & wait for graceful exit.
-	require.NoError(t, schedulerPool.Wait())
 	cancel()
+	require.NoError(t, schedulerPool.Wait())
 }
 
 func TestScheduler_Committee_Indices_Changed_Attester_Only_2(t *testing.T) {

--- a/operator/duties/committee_test.go
+++ b/operator/duties/committee_test.go
@@ -2,6 +2,7 @@ package duties
 
 import (
 	"context"
+	"runtime"
 	"testing"
 	"time"
 
@@ -379,6 +380,11 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only(t *testing.T) {
 	currentSlot.Set(phase0.Slot(1))
 	waitForDuties.Set(true)
 	ticker.Send(currentSlot.Get())
+	// Wait for the slot ticker to be triggered in the attester, sync committee, and cluster handlers.
+	// This ensures that no attester duties are fetched before the cluster ticker is triggered,
+	// preventing a scenario where the cluster handler executes duties in the same slot as the attester fetching them.
+	time.Sleep(10 * time.Millisecond)
+
 	// wait for attester duties to be fetched
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 	// wait for sync committee duties to be fetched
@@ -477,6 +483,11 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only_2(t *testing.T) {
 	currentSlot.Set(phase0.Slot(1))
 	waitForDuties.Set(true)
 	ticker.Send(currentSlot.Get())
+	// Wait for the slot ticker to be triggered in the attester, sync committee, and cluster handlers.
+	// This ensures that no attester duties are fetched before the cluster ticker is triggered,
+	// preventing a scenario where the cluster handler executes duties in the same slot as the attester fetching them.
+	time.Sleep(10 * time.Millisecond)
+
 	// wait for attester duties to be fetched
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 	// wait for sync committee duties to be fetched
@@ -565,6 +576,11 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only_3(t *testing.T) {
 	currentSlot.Set(phase0.Slot(1))
 	waitForDuties.Set(true)
 	ticker.Send(currentSlot.Get())
+	// Wait for the slot ticker to be triggered in the attester, sync committee, and cluster handlers.
+	// This ensures that no attester duties are fetched before the cluster ticker is triggered,
+	// preventing a scenario where the cluster handler executes duties in the same slot as the attester fetching them.
+	time.Sleep(10 * time.Millisecond)
+
 	// wait for attester duties to be fetched
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 	// wait for sync committee duties to be fetched
@@ -592,6 +608,8 @@ func TestScheduler_Committee_Indices_Changed_Attester_Only_3(t *testing.T) {
 
 // reorg previous dependent root changed
 func TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only(t *testing.T) {
+	originalProcs := runtime.GOMAXPROCS(1)
+	defer runtime.GOMAXPROCS(originalProcs)
 	var (
 		dutyStore     = dutystore.New()
 		attHandler    = NewAttesterHandler(dutyStore.Attester)
@@ -666,6 +684,8 @@ func TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Attester_only(t *te
 	scheduler.HandleHeadEvent(logger)(e)
 	// wait for attester duties to be fetched again for the current epoch
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
+	// no execution should happen in slot 64
+	waitForNoActionCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
 	// STEP 5: execute reorged duty
 	currentSlot.Set(phase0.Slot(65))
@@ -770,6 +790,8 @@ func TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Indices_Changed_Att
 	scheduler.HandleHeadEvent(logger)(e)
 	// wait for attester duties to be fetched
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
+	// no execution should happen in slot 64
+	waitForNoActionCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
 	// STEP 5: trigger indices change
 	scheduler.indicesChg <- struct{}{}
@@ -779,6 +801,11 @@ func TestScheduler_Committee_Reorg_Previous_Epoch_Transition_Indices_Changed_Att
 	// STEP 6: wait for attester duties to be fetched again for the current epoch
 	currentSlot.Set(phase0.Slot(65))
 	ticker.Send(currentSlot.Get())
+	// Wait for the slot ticker to be triggered in the attester, sync committee, and cluster handlers.
+	// This ensures that no attester duties are fetched before the cluster ticker is triggered,
+	// preventing a scenario where the cluster handler executes duties in the same slot as the attester fetching them.
+	time.Sleep(10 * time.Millisecond)
+
 	// wait for attester duties to be fetched
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 	// wait for sync committee duties to be fetched
@@ -870,6 +897,8 @@ func TestScheduler_Committee_Reorg_Previous_Attester_only(t *testing.T) {
 	scheduler.HandleHeadEvent(logger)(e)
 	// wait for attester duties to be fetched again for the current epoch
 	waitForDutiesFetchCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
+	// no execution should happen in slot 33
+	waitForNoActionCommittee(t, logger, fetchDutiesCall, executeDutiesCall, timeout)
 
 	// STEP 5: Ticker with no action
 	currentSlot.Set(phase0.Slot(34))

